overlord/snapstate/snapmgr.go

```diff
@@ -107,6 +111,7 @@ type SnapState struct {
 	Active    bool             `json:"active,omitempty"`
 	Channel   string           `json:"channel,omitempty"`
 	Flags     SnapStateFlags   `json:"flags,omitempty"`
+	RollbackR snap.Revision    `json:"rollback,omitempty"`
```

Comment:
if we think we'll need to support more than one rollback later, maybe this should start as []snap.Revision from day 1? to avoid the UnmarshalJSON juggling, though it wouldn't be too bad either